flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSinkUtils.java

```diff
@@ -209,6 +213,78 @@ public static RelNode convertSinkToRel(
                 sink);
     }
 
+    /**
+     * Converts a given {@link DynamicTableSink} to a {@link RelNode}. It adds helper projections if
+     * necessary.
+     */
+    public static RelNode convertCtasToRel(
+            FlinkRelBuilder relBuilder,
+            RelNode input,
+            Catalog catalog,
+            CreateTableASOperation createTableASOperation,
+            DynamicTableSink sink) {
+        final ResolvedCatalogTable catalogTable =
+                (ResolvedCatalogTable)
+                        createTableASOperation.getCreateTableOperation().getCatalogTable();
+
+        final ObjectIdentifier identifier =
+                createTableASOperation.getCreateTableOperation().getTableIdentifier();
+
+        final ContextResolvedTable contextResolvedTable;
+        if (createTableASOperation.getCreateTableOperation().isTemporary()) {
+            contextResolvedTable = ContextResolvedTable.temporary(identifier, catalogTable);
+        } else {
+            contextResolvedTable =
+                    ContextResolvedTable.permanent(identifier, catalog, catalogTable);
+        }
+
+        return convertSinkToRel(
+                relBuilder,
+                input,
+                Collections.emptyMap(),
+                contextResolvedTable,
+                createTableASOperation.getSinkModifyStaticPartitions(),
+                null,
+                createTableASOperation.getSinkModifyOverwrite(),
+                sink);
+    }
+
+    /**
+     * Converts a given {@link DynamicTableSink} to a {@link RelNode}. It adds helper projections if
+     * necessary.
+     */
+    public static RelNode convertRtasToRel(
+            FlinkRelBuilder relBuilder,
+            RelNode input,
+            Catalog catalog,
+            ReplaceTableAsOperation replaceTableASOperation,
+            DynamicTableSink sink) {
+        final ResolvedCatalogTable catalogTable =
+                (ResolvedCatalogTable)
+                        replaceTableASOperation.getCreateTableOperation().getCatalogTable();
+
+        final ObjectIdentifier identifier =
+                replaceTableASOperation.getCreateTableOperation().getTableIdentifier();
+
+        final ContextResolvedTable contextResolvedTable;
+        if (replaceTableASOperation.getCreateTableOperation().isTemporary()) {
```

Comment:
pass `replaceTableASOperation.getCreateTableOperation()` as a parameter and merge `convertRtasToRel` and `convertCtasToRel` to reduce code duplication